lib/ecto/adapters/postgres/connection.ex

```diff
@@ -409,8 +409,9 @@ if Code.ensure_loaded?(Postgrex) do
 
     defp cte(%{with_ctes: _}, _), do: []
 
-    defp cte_expr({name, cte}, sources, query) do
-      [quote_name(name), " AS ", cte_query(cte, sources, query)]
+    defp cte_expr({name, materialize, cte}, sources, query) do
+      materialize_opt = if materialize, do: "MATERIALIZED ", else: ""
+      [quote_name(name), " AS ", materialize_opt, cte_query(cte, sources, query)]
```

Comment:
```suggestion
    defp cte_expr({name, materialized, cte}, sources, query) do
      materialized_opt =
        case materialized do
          nil -> ""
          true -> "MATERIALIZED"
          false -> "NOT MATERIALIZED"  
        end
        
        [quote_name(name), " AS ", materialized_opt, cte_query(cte, sources, query)]
```

Based on what we talked about in the ecto PR. also the variable is `materliazed` instead of `materialize` so better to be consistent